calico_containers/calico_ctl/pool.py

```diff
@@ -95,24 +112,56 @@ def ip_pool_add(cidr_pool, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    cidr = check_ip_version(cidr_pool, version, IPNetwork)
-    pool = IPPool(cidr, ipip=ipip, masquerade=masquerade)
-    client.add_ip_pool(version, pool)
+    # TODO Reject any cidrs that overlap with existing cidrs in the pool
+    for cidr in cidrs:
+        pool = IPPool(cidr, ipip=ipip, masquerade=masquerade)
+        client.add_ip_pool(version, pool)
+
+
+def ip_pool_range_add(start_ip, end_ip, version, ipip, masquerade):
+    """
+    Add the range of ip addresses as CIDRs to the IP address allocation pool.
+
+    :param start_ip: The first ip address the ip range.
+    :param end_ip: The last ip address in the ip range.
+    :param version: v4 or v6
+    :param ipip: Use IP in IP for this pool.
+    :return: None
+    """
+    if version == "v6" and ipip:
+        print "IP in IP not supported for IPv6 pools"
+        sys.exit(1)
+
+    ip_range = IPRange(start_ip, end_ip)
+    pools = client.get_ip_pools(version)
+    for pool in pools:
+        pool_net = IPNetwork(pool.cidr)
+        # If the ip range intersects with any existing ip pool, reject it
```

Comment:
This is a slightly complicated comparison. Can you add a bit more explanation in the comment.